server/routes/polls/polls.route.js

```diff
@@ -0,0 +1,87 @@
+const express = require("express");
+const router = express.Router();
+const Poll = require("../../models/polls");
+const upload = require("./utils");
+
+// @route POST /polls
+// @desc To add a new poll
+// @access private
+// @params title, userId, imagess x 2
+router.post("/", upload.any(), (req, res) => {
+  const errors = [];
+  if (!req.body.title) errors.push({ title: "Invalid title" });
+  if (!req.body.userId) errors.push({ userId: "Invalid user id" });
+  if (req.files.length != 2)
+    errors.push({ images: "minimum two images required" });
+
+  if (errors.length > 0) return res.status(400).json({ error: errors });
+
+  const poll = new Poll({
+    title: req.body.title,
+    userId: req.body.userId,
+  });
+  // Trying to save the poll
+  try {
+    poll.save();
+  } catch (error) {
+    return res.status(400).json({ error: error.message });
+  }
+  // Saving file location to the poll object
+  try {
+    req.files.map((item) => {
+      console.log(item);
+      poll.addImage(item.location);
+    });
+  } catch (error) {
+    return res.status(400).json({ error: error.message });
+  }
+  return res.status(200).json({ poll: poll });
+});
+
+// @route DELETE "/polls"
+// @desc to be delete a poll with a given pollId
+// @params pollId
+// @access private
+router.delete("/", (req, res) => {
+  const pollId = req.body.id;
+  Poll.findByIdAndDelete(pollId, (error) => {
+    if (error) return res.status(400).send(error);
+    else {
+      return res.status(200).json({ status: "ok" });
+    }
+  });
+});
```

Comment:
When I tested it, this has returned `{ status: "ok" }` even when I entered invalid IDs.

It looks like it's returning `error` as `null` even when I enter invalid ids.

Maybe something like [`remove()`](https://mongoosejs.com/docs/api.html#model_Model-remove) may work